### ISSUE TITLE
Not able to parse dynamic import

### ISSUE THREAD

**Issue by MaxMilton** [ampproject/rollup-plugin-closure-compiler#48]
## What's the issue?

In trying to add this package in <https://github.com/MaxMilton/sapper-template-rollup/tree/feat/use-closure-compiler> I ran into the issue of `import()` not able to be be parsed processed by acorn, which is used through this package.

## How do we reproduce the issue?

1. Clone repo branch <https://github.com/MaxMilton/sapper-template-rollup/tree/feat/use-closure-compiler>.
1. Install deps; `yarn install`.
1. Run build; `yarn run build`.

Result:

```
yarn run v1.9.4
warning package.json: No license field
$ sapper build
> Building...
> Error transforming chunk with 'closure-compiler' plugin: baseVisitor[type] is not a function
error Command failed with exit code 1.
```

After some debugging you'll find acorn is trying to do `baseVisitor['Import']`, which is not available in acorn at the moment.

## Additional context

By adding the `acorn-dynamic-import` package it's possible to support parsing `import()`. In fact this is exactly what [closure-webpack-plugin](https://github.com/webpack-contrib/closure-webpack-plugin/blob/db6ba8106b9e98f3f8d56b97bca76b70a3b4fab4/src/goog-require-parser-plugin.js#L3) is doing, as well as others like [buble](https://github.com/Rich-Harris/buble/pull/102/files).

One other critical part of this is support in Closure Compiler itself for parser support for dynamic import: <https://github.com/google/closure-compiler/issues/2770>. Although, it is possible for a rollup plugin to rewrite the `import()` into something else before it ever hits Closure so this issue is not necessarily dependant upon Closure parser support.

**Comment by kristoferbaxter** [comment]
Happy to move over, I planned to move to the next major version of Acorn with the next release but will attempt to address this sooner.

**Comment by kristoferbaxter** [comment]
If anyone else wants to take a look at this before I have a chance to return to it, here's the branch with the progress I made.

https://github.com/ampproject/rollup-plugin-closure-compiler/tree/dynamic-import

**Comment by kristoferbaxter** [comment]
Opened an issue on the `acorn` repo, https://github.com/acornjs/acorn/issues/746

It appears that `acorn-walk` doesn't fully function when injecting plugins.

**Comment by kristoferbaxter** [comment]
Created a repo with workaround for `acorn-walk` not working as expected with plugin injection: https://github.com/kristoferbaxter/dynamic-walk

Found another root issue preventing this from working: Each Plugin invocation creates a singluar set of transforms and applies them to source. When `experimentalCodeSplitting` is enabled in `rollup`, the plugin is invoked once causing the singleton transforms to be used on multiple outputs.

This means a parsed AST for one file is used to perform transformations on others. Whoops!

**Comment by kristoferbaxter** [comment]
Released `0.8.0` with dynamic import support for simple optimizations.

**Comment by mckravchyk** [comment]
@kristoferbaxter I'm getting the same error message on 0.27.0 (simple optimizations) when I try to use MobX, points to `acorn-walk/dist/walk.js:29:24` -
